backend/src/services/registerService.js

```diff
@@ -0,0 +1,32 @@
+import { registerData } from '../repos/registerData';
+
+export const filterInput = (input) => {
+    if ( !input.username && !input.password ) return 'Username and password are required.';
+    if ( input.username && !input.password ) return 'Password is required.';
+    if ( !input.username && input.password ) return 'Username is required.';
+    if ( input.password.length < 8 ) return 'Password must be 8 characters.';
+    return '';
+};
+
+export const registerService = (input) => {
+    return new Promise( async (resolve,reject) => {
+        const invalidInput = filterInput(input);
+        if ( invalidInput ) {
+            reject(invalidInput);
+        } else {
+            try {
+                const userid = await registerData('user',[input.username,input.password]);
```

Comment:
@gulyaasjaanos `db.save('user', {username: input.username, password: input.password})`
- db should check if the user to be saved is valid